### ISSUE TITLE
MacOS and Android Issue: SocketException: Connection refused erno = 111

### ISSUE THREAD

**Issue by stardomains3** [Mobile-Artificial-Intelligence/maid#289]
On MacOS 14.2.1
"Ollama serve" entered in Terminal. Menubar app not running.
MAID 1.1.8 on Pixel 8 Pro Android
All Permissions granted
Disabled firewall too
Both are on same Wifi network

> ollama serve
> 2024/01/17 14:12:47 images.go:808: total blobs: 15
> 2024/01/17 14:12:47 images.go:815: total unused blobs removed: 0
> 2024/01/17 14:12:47 routes.go:930: Listening on 127.0.0.1:11434 (version 0.1.20)

In Maid: Model-> API Type: Ollama -> Remote URL http://127.0.0.1:11434

No Remote Models load in the dropdown (I have three on my MacBook Air that I use on the Mac normally)

Settings Log: 

> Model created with name: null
> Character created with name: Maid
> Nearby Devices - Permission granted
> Error: ClientException with SocketException: Connection refused (OS Error: Connection refused, errno = 111), address = 127.0.0.1, port = 35932, uri=http://127.0.0.1:11434/api/tags
> Nearby Devices - Permission granted
> Error: ClientException with SocketException: Connection refused (OS Error: Connection refused, errno = 111), address = 127.0.0.1, port = 35356, uri=http://127.0.0.1:11434/api/tags
> Model Saved:

Am I doing anything incorrectly? Is it looking on the wrong port: "address = 127.0.0.1, port = 35356" ? IDK I've tried many times but it seems to look at ports other than 11434.

**Comment by danemadsen** [comment]
I have no way of testing on MacOS but ill look at it on android

**Comment by danemadsen** [comment]
You can check if ollama is serving correctly by going to http://127.0.0.1:11434/api/tags in your browser. http://127.0.0.1:11434 is the default address ollama serves to but if you've set it to serve over the network it will be set to http://0.0.0.0:11434 on the machine ollama is served or the ip address of that machine on the other devices in your network.

**Comment by stardomains3** [comment]
At http://127.0.0.1:11434/api/tags :

> {"models":[{"name":"llama2:latest","modified_at":"2024-01-12T09:02:03.117433522-05:00","size":3826793677,"digest":"78e26419b4469263f75331927a00a0284ef6544c1975b826b15abdaef17bb962","details":{"format":"gguf","family":"llama","families":["llama"],"parameter_size":"7B","quantization_level":"Q4_0"}},{"name":"mistral:latest","modified_at":"2024-01-17T15:54:51.299049204-05:00","size":4109865159,"digest":"61e88e884507ba5e06c49b40e6226884b2a16e872382c2b44a42f2d119d804a5","details":{"format":"gguf","family":"llama","families":["llama"],"parameter_size":"7B","quantization_level":"Q4_0"}},{"name":"openhermes:latest","modified_at":"2024-01-12T19:13:50.588834664-05:00","size":4108928574,"digest":"95477a2659b7539758230498d6ea9f6bfa5aa51ffb3dea9f37c91cacbac459c1","details":{"format":"gguf","family":"llama","families":["llama"],"parameter_size":"7B","quantization_level":"Q4_0"}},{"name":"wizard-vicuna-uncensored:13b","modified_at":"2024-01-17T16:17:13.734950815-05:00","size":7365835082,"digest":"6887722b661837283fb4409beb9b6eb155e2146770172ad2392d9b707ff5e9fa","details":{"format":"gguf","family":"llama","families":null,"parameter_size":"13B","quantization_level":"Q4_0"}},{"name":"wizardlm-uncensored:latest","modified_at":"2024-01-12T19:29:52.142674653-05:00","size":7365835187,"digest":"886a369d74fcaeb3d9293e81c1de8cb87a5142521f845447c780f05d9439607a","details":{"format":"gguf","family":"llama","families":null,"parameter_size":"13B","quantization_level":"Q4_0"}}]}

log:

> Model Initialised
Character Initialised
Session Initialised
Model created with name: null
{chatml: false, instruct: false, interactive: true, random_seed: true, path: , seed: -1, n_ctx: 512, n_batch: 8, n_threads: 8, n_predict: 512, n_keep: 48, n_prev: 64, n_probs: 0, top_k: 40, top_p: 0.95, min_p: 0.1, tfs_z: 1.0, typical_p: 1.0, temperature: 0.8, penalty_last_n: 64, penalty_repeat: 1.1, penalty_freq: 0.0, penalty_present: 0.0, mirostat: 0, mirostat_tau: 5.0, mirostat_eta: 0.1, penalize_nl: true, remote_url: http://127.0.0.1:11434/, api_type: 3, preset: Default}
Character created with name: Maid
Nearby Devices - Permission granted
Error: ClientException with SocketException: Connection refused (OS Error: Connection refused, errno = 111), address = 127.0.0.1, port = 59520, uri=http://127.0.0.1:11434/api/tags
Nearby Devices - Permission granted
Nearby Devices - Permission granted
Nearby Devices - Permission granted
Nearby Devices - Permission granted
Error: ClientException with SocketException: Failed host lookup: '127.0.0' (OS Error: No address associated with hostname, errno = 7), uri=[http://127.0.0:11434/api/tags](http://127.0.0.0:11434/api/tags)
Error: ClientException with SocketException: Failed host lookup: '127.0.' (OS Error: No address associated with hostname, errno = 7), uri=[http://127.0.:11434/api/tags](http://127.0.0.0:11434/api/tags)
Error: ClientException with SocketException: Failed host lookup: '127.0.0.' (OS Error: No address associated with hostname, errno = 7), uri=[http://127.0.0.:11434/api/tags](http://127.0.0.0:11434/api/tags)
Nearby Devices - Permission granted
Error: ClientException with SocketException: Failed host lookup: '127.0' (OS Error: No address associated with hostname, errno = 7), uri=[http://127.0:11434/api/tags](http://127.0.0.0:11434/api/tags)
Nearby Devices - Permission granted
Error: ClientException with SocketException: Failed host lookup: '127.' (OS Error: No address associated with hostname, errno = 7), uri=[http://127.:11434/api/tags](http://0.0.0.127:11434/api/tags)
Nearby Devices - Permission granted
Error: ClientException with SocketException: Failed host lookup: '127' (OS Error: No address associated with hostname, errno = 7), uri=[http://127:11434/api/tags](http://0.0.0.127:11434/api/tags)
Error: ClientException with SocketException: Failed host lookup: '12' (OS Error: No address associated with hostname, errno = 7), uri=[http://12:11434/api/tags](http://0.0.0.12:11434/api/tags)
Nearby Devices - Permission granted
Nearby Devices - Permission granted
Error: Invalid argument(s): No host specified in URI http://:11434/api/tags
Error: ClientException with SocketException: Failed host lookup: '1' (OS Error: No address associated with hostname, errno = 7), uri=[http://1:11434/api/tags](http://0.0.0.1:11434/api/tags)
Nearby Devices - Permission granted
Error: ClientException with SocketException: Failed host lookup: '1' (OS Error: No address associated with hostname, errno = 7), uri=[http://1:11434/api/tags](http://0.0.0.1:11434/api/tags)
Nearby Devices - Permission granted
Error: ClientException with SocketException: Failed host lookup: '10' (OS Error: No address associated with hostname, errno = 7), uri=[http://10:11434/api/tags](http://0.0.0.10:11434/api/tags)
Nearby Devices - Permission granted
Error: ClientException with SocketException: Failed host lookup: '10.' (OS Error: No address associated with hostname, errno = 7), uri=[http://10.:11434/api/tags](http://0.0.0.10:11434/api/tags)
Nearby Devices - Permission granted
Error: ClientException with SocketException: Failed host lookup: '10.0' (OS Error: No address associated with hostname, errno = 7), uri=[http://10.0:11434/api/tags](http://10.0.0.0:11434/api/tags)
Nearby Devices - Permission granted
Error: ClientException with SocketException: Failed host lookup: '10.0.' (OS Error: No address associated with hostname, errno = 7), uri=[http://10.0.:11434/api/tags](http://10.0.0.0:11434/api/tags)
Nearby Devices - Permission granted
Error: ClientException with SocketException: Failed host lookup: '10.0.0' (OS Error: No address associated with hostname, errno = 7), uri=[http://10.0.0:11434/api/tags](http://10.0.0.0:11434/api/tags)
Nearby Devices - Permission granted
Error: ClientException with SocketException: Failed host lookup: '10.0.0.' (OS Error: No address associated with hostname, errno = 7), uri=[http://10.0.0.:11434/api/tags](http://10.0.0.0:11434/api/tags)
Nearby Devices - Permission granted
Nearby Devices - Permission granted
Error: ClientException with SocketException: Connection refused (OS Error: Connection refused, errno = 111), address = 10.0.0.39, port = 57038, uri=http://10.0.0.39:11434/api/tags
Error: ClientException with SocketException: No route to host (OS Error: No route to host, errno = 113), address = 10.0.0.3, port = 34570, uri=http://10.0.0.3:11434/api/tags
Model Saved: 
Nearby Devices - Permission granted
Error: ClientException with SocketException: Connection refused (OS Error: Connection refused, errno = 111), address = 10.0.0.39, port = 39376, uri=http://10.0.0.39:11434/api/tags
Nearby Devices - Permission granted
Error: ClientException with SocketException: Connection refused (OS Error: Connection refused, errno = 111), address = 10.0.0.39, port = 42850, uri=http://10.0.0.39:1143/api/tags
Nearby Devices - Permission granted
Error: ClientException with SocketException: Connection refused (OS Error: Connection refused, errno = 111), address = 10.0.0.39, port = 39388, uri=http://10.0.0.39:11434/api/tags
Model Saved: 
Nearby Devices - Permission granted
Error: ClientException with SocketException: Connection refused (OS Error: Connection refused, errno = 111), address = 10.0.0.39, port = 40748, uri=http://10.0.0.39:11434/api/tags
Nearby Devices - Permission granted
Nearby Devices - Permission granted
Nearby Devices - Permission granted
Nearby Devices - Permission granted
Error: ClientException with SocketException: Connection refused (OS Error: Connection refused, errno = 111), address = 0.0.0.0, port = 37214, uri=http://0.0.0.0:11434/api/tags
Error: ClientException with SocketException: Failed host lookup: '0.0.0.' (OS Error: No address associated with hostname, errno = 7), uri=[http://0.0.0.:11434/api/tags](http://0.0.0.0:11434/api/tags)
Model Saved:

I tried:
http://127.0.0.1:11434/ "Ollama is running" on browser on MacBook
http://0.0.0.0:11434/  "Ollama is running" on browser on MacBook
http://10.0.0.39:11434/  MacBook IP

**Comment by danemadsen** [comment]
So you tried your macbook IP on another device? If it doesnt show Ollama is running" it probably means ollama isnt set up correctly. make sure you arnt using anything like a vpn or perhaps theres a firewall on your router

**Comment by stardomains3** [comment]
> So you tried your macbook IP on another device? If it doesnt show Ollama is running" it probably means ollama isnt set up correctly. make sure you arnt using anything like a vpn or perhaps theres a firewall on your router

OK. Thanks for the help. Will keep researching. My goal is just trying to get it to work on the same network. MacBook serving Ollama and Maid working with it remotely.

**Comment by danemadsen** [comment]
Yeah sorry i cant help more, apple support is very limited at the moment, i really need another experienced developer that can help with that side of things.

**Comment by stardomains3** [comment]
By the way, I solved this so I can run Ollama on Maid on my local network. I found the IP of my MacBook. Then added that as the OLLAMA_HOST IP. In Terminal I run this command: `OLLAMA_ORIGINS=https://10.0.0.11 OLLAMA_HOST=10.0.0.32:8080 ollama serve` Where the origins IP is that of the Android device running MAID.

Then in MAID Model select Ollama. Remote URL: 10.0.0.32:8080 ... and Voila! The models populated in the dropdown box and I can converse with my MacBook Ollama serve models!
